lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -299,13 +299,13 @@ def test_http_exception(
         if not self.elastic_endpoint:
             pytest.skip("skipping " + self.test_http_exception.__name__)
 
-        if self.payload.get("start") and self.payload.get("end"):
+        if not self.use_index_from_metadata:
             index = self.build_index(
                 server_config,
                 self.date_range(self.payload["start"], self.payload["end"]),
             )
         else:
-            index = self.build_index_from_metadata(provide_metadata)
+            index = self.build_index_from_metadata()
```

Comment:
This code would be prettier if you reversed the clauses:
```
        if self.use_index_from_metadata:
            index = self.build_index_from_metadata()
        else:
            index = self.build_index(
                server_config,
                self.date_range(self.payload["start"], self.payload["end"]),
            )
```

Reply:
Ditto lines 338-344.

I kind of wonder whether you should refactor this code into a helper function, or even put it back inside `build_index()`, now that you have a nice way of handling the build-from-metadata case.